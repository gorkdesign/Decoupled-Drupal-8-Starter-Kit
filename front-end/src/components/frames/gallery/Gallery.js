import React from 'react';
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';

import Thumbnail from './Thumbnail';
import BrowseButton from './BrowseButton';

const Gallery = ({
  onDrop,
  uploading,
  files,
  totalBytes,
  uploadInitiated = false,
  onUploadClick,
  handleCancel,
  handleDelete,
}) => {
  let dropzoneRef;

  return (
    <div className="GalleryFrameContainer">

      { uploading === false ?
        <Dropzone
          ref={(node) => { dropzoneRef = node; }}
          onDrop={onDrop}
          id="dropZone"
          className="dropZone"
          disabled={uploading}
        >
          <p>Drop your files here or click to browse.</p>
        </Dropzone>
      :
        <div id="dropZone" className="dropZone disabled">
          <p>Your files are uploading.</p>
        </div>
      }

      {uploading === false ?
        <BrowseButton
          files={files}
          totalBytes={totalBytes(files)}
          totalFiles={files.length}
          render={() => (
            <button
              type="button"
              onClick={() => { dropzoneRef.open(); }}
              disabled={uploadInitiated}
            >
            Choose Files
            </button>
        )}
        />
      : null }

      { files.length > 0 ?
        <button className="btn btn-primary" onClick={onUploadClick}>UPLOAD FILES</button>
      : null }

      <output id="list" className="container p-0">
        <div className="grid">
          {
            files.map((item, i) => {
              const image = files[i].file;
              if (image) {
                return (<Thumbnail
                  key={i}
                  handleCancel={handleCancel}
                  handleDelete={handleDelete}
                  index={i}
                  fileSize={image.size || image.fileSize}
                  fileName={image.name}
                  percentageComplete={image.percentCompleted ? image.percentCompleted : 0}
                  uploadInitiated={image.uploadInitiated ? image.uploadInitiated : false}
                  uploadSuccess={image.uploadSuccess ? image.uploadSuccess : false}
                  render={() => (
                    <figure>
                      <img alt="" src={item.thumbnail} className="responsive-image" />
                    </figure>
                  )}
                />);
              }
              return null;
            })
          }
        </div>
      </output>
    </div>
  );
};

Gallery.propTypes = {
  uploading: PropTypes.bool.isRequired,
  files: PropTypes.array.isRequired,
  uploadInitiated: PropTypes.bool,
  onDrop: PropTypes.func.isRequired,
  totalBytes: PropTypes.func.isRequired,
  onUploadClick: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

Gallery.defaultProps = {
  uploadInitiated: false,
};

export default Gallery;
