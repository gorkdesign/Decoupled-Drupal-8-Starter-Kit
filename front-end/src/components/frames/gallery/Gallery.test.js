import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import Gallery from './Gallery';

const props = {
  uploading: false,
  files: [],
  uploadInitiated: false,
  onDrop: jest.fn(),
  computedTotalBytes: jest.fn(),
  onUploadClick: jest.fn(),
  handleCancel: jest.fn(),
  handleDelete: jest.fn(),
};

const generateShallow = (props) => {
  return shallow(
    <Gallery {...props} />
  );
}


describe('Gallery', () => {

  it('should render correctly', () => {
    const output = generateShallow(props);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

});