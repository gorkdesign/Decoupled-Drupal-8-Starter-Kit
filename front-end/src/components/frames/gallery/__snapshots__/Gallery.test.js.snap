// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should render correctly 1`] = `
<div
  className="GalleryFrameContainer"
>
  <Dropzone
    className="dropZone"
    disableClick={false}
    disablePreview={false}
    disabled={false}
    id="dropZone"
    maxSize={Infinity}
    minSize={0}
    multiple={true}
    onDrop={[Function]}
    preventDropOnDocument={true}
  >
    <p>
      Drop your files here or click to browse.
    </p>
  </Dropzone>
  <BrowseButton
    files={Array []}
    render={[Function]}
    totalBytes={0}
    totalFiles={0}
  />
  <output
    className="container"
    id="list"
  >
    <div
      className="grid"
    />
  </output>
</div>
`;
